### PR TITLE
ARC-99: Viewer

### PR DESCRIPTION
**Pull request by tjboswell**
rough implementation of web viewer. pdf viewer is implemented but does not get pdfs from backend yet (change activeItemContentType in item-reducers to 'pdf' to test pdf viewer).

### CODE REVIEW

File: src/state/item/item-reducers.js [hunk 1]
```diff
@@ -13,6 +13,8 @@ const initialState = {
     activeItemIndex: null,
     activeItemIndexCached: null, // saves the index of active item on different page
     activeItemPage: null,
+    activeItemContent: null,
+    activeItemContentType: 'web', // TODO: figure out content type from response
```
Comment (mr2ert) [inline src/state/item/item-reducers.js]: Check the response headers -- should be under `Content-Type`.  Also, I'd double check when you pull a pdf it doesn't attempt to download it in a way the user can interact with (this is what postman does for me).

Comment (mr2ert) [reply src/state/item/item-reducers.js]: http://stackoverflow.com/a/34430105/2336149

**Comment by tjboswell** [comment]
should be good to test. things like handling links and getting numPages for pdfs need to be done, but html and pdf files should render correctly.

**Comment by mr2ert** [comment]
@tjboswell What is the status of this?  Can we merge it in?

**Comment by tjboswell** [comment]
If someone has looked at it and tested it, then yes it should be ready to merge

**Comment by mr2ert** [comment]
Bugs/features:
1) The PDF viewer only shows the first 3 pages -- even when there are way more.
2) On load, most of the time the PDFs and HTMLs start out all highlighted (in chrome this looks like a blue box over the page for html, and blue highlighting for the text in pdfs) -- this looks bad and is a little confusing.
3) I'd really like to eventually get continuous scroll working.
4) Eventually, I'd like a spinner to appear while the content is loading.

That said, I really want to get this in for now -- it is more than enough of a working version. :shipit: @ScottSelinger can you make tickets for these things?

**Comment by ScottSelinger** [comment]
Yep will do later tonight.

**Pull request merged**